lib/src/sql/function.rs

```diff
@@ -161,6 +165,40 @@ impl fmt::Display for Function {
 	}
 }
 
+impl Serialize for Function {
+	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+	where
+		S: serde::Serializer,
+	{
+		if is_internal_serialization() {
+			match self {
+				Self::Cast(s, e) => {
+					let mut serializer = serializer.serialize_tuple_variant(TOKEN, 0, "Cast", 2)?;
+					serializer.serialize_field(s)?;
+					serializer.serialize_field(e)?;
+					serializer.end()
+				}
+				Self::Normal(s, e) => {
+					let mut serializer =
+						serializer.serialize_tuple_variant(TOKEN, 1, "Normal", 2)?;
+					serializer.serialize_field(s)?;
+					serializer.serialize_field(e)?;
+					serializer.end()
+				}
+				Self::Script(s, e) => {
+					let mut serializer =
+						serializer.serialize_tuple_variant(TOKEN, 2, "Script", 2)?;
+					serializer.serialize_field(s)?;
+					serializer.serialize_field(e)?;
+					serializer.end()
+				}
+			}
+		} else {
+			serializer.serialize_some(&self.to_string())
```

Comment:
We should never get to serialising this type as a basic JSON/CBOR output, as we will always have called `compute` on this value by this time, and converted it to a primary type (Null, False, True, Number, Strand, Duration, Datetime, Uuid, Array, Object, Geometry). Therefore maybe a `serialize_none()` is better here?